packages/tx/src/eip2930Transaction.ts

```diff
@@ -188,6 +188,10 @@ export default class AccessListEIP2930Transaction extends BaseTransaction<Access
 
     this.gasPrice = new BN(toBuffer(gasPrice === '' ? '0x' : gasPrice))
 
+    if (this.gasPrice.mul(this.gasLimit).toArray().length > 32) {
+      throw new Error('gas limit * price overflow')
```

Comment:
is this specified somewhere?

Reply:
Here again, not specified anywhere but if the length of the array version of the BN is greater than 32, that would imply it's greater than the greatest possible integer that Ethereum handles, right?  [This test](https://github.com/ethereum/tests/blob/develop/TransactionTests/ttGasLimit/TransactionWithGasLimitxPriceOverflow.json) expects this to be an overflow error and was failing before this change.